cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/services/ApplicationLifecycleService.java

```diff
@@ -630,6 +650,20 @@ private void deleteAppMetadata(Id.Application appId, ApplicationSpecification ap
   }
 
   /**
+   * Ensures that the logged-in user has a {@link Action privilege} on the specified dataset instance.
+   *
+   * @param appId the {@link ApplicationId} to check for privileges
+   * @throws UnauthorizedException if the logged in user has no {@link Action privileges} on the specified dataset
+   */
+  private void ensureAccess(ApplicationId appId) throws Exception {
+    Principal principal = authenticationContext.getPrincipal();
+    co.cask.cdap.api.Predicate<EntityId> filter = authorizationEnforcer.createFilter(principal);
+    if (!Principal.SYSTEM.equals(principal) && !filter.apply(appId)) {
+      throw new UnauthorizedException(principal, Action.READ, appId);
```

Comment:
Why only for Read Action? If it checks only for read, can we rename the method to ensureReadAccess?